From dca8e827c1c9450f809f763a816394d66b7b3097 Mon Sep 17 00:00:00 2001
From: sb6596 <shubhamprince111@gmail.com>
Date: Sat, 1 Dec 2018 07:16:22 +0000
Subject: [PATCH 4/5] Allow fsck_untrusted to getattr block_device

 * Noted upon conntecing usb_otg
 * avc: denied { getattr } for path="/dev/block" dev="tmpfs"
   ino=8466 scontext=u:r:fsck_untrusted:s0
   tcontext=u:object_r:block_device:s0 tclass=dir permissive=0

Change-Id: I0b95c3ee1b76f6bf0ecc030a628baab95e65861b
---
 public/fsck_untrusted.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/public/fsck_untrusted.te b/public/fsck_untrusted.te
index 8510c942..80f9ab33 100644
--- a/public/fsck_untrusted.te
+++ b/public/fsck_untrusted.te
@@ -9,7 +9,7 @@ allow fsck_untrusted vold:fd use;
 allow fsck_untrusted vold:fifo_file { read write getattr };
 
 # Run fsck on vold block devices
-allow fsck_untrusted block_device:dir search;
+allow fsck_untrusted block_device:dir { getattr search };
 allow fsck_untrusted vold_device:blk_file rw_file_perms;
 
 allow fsck_untrusted proc_mounts:file r_file_perms;
-- 
2.17.1

